### DISCUSSION TITLE
Text Color and Text Size classes have issues

### DISCUSSION THREAD

**Discussion by jahangirdev** [tailwindlabs/tailwindcss#14572] [Help]
"text-white" defines text color.
"text-xl" defines text size.
Both of them are started with "text-" prefix.

For custom text colors we can use "text-[#ffffff]". It works fine.
For custom text color using css variable we can use "text-[var(--text-color)]". It also works fine.

For custom text size we can use "text-[26px]". It also works fine.
But when I use the custom font size using css variable "text-[var(--text-size)]". It doesn't work.

### It renders the CSS like below:

```
.text-\[var\(--text-size\)\] {
    color: var(--text-size);
}
```

### But it should be:

```
.text-\[var\(--text-size\)\] {
    font-size: var(--text-size);
}
```

**Comment by adamwathan** [comment]
Hey! This exact situation is explained in the docs with a solution here:

https://tailwindcss.com/docs/adding-custom-styles#resolving-ambiguities
